sktime/datatypes/_adapter/polars.py

```diff
@@ -119,6 +143,47 @@ def check_polars_frame(obj, return_metadata=False, var_name="obj", lazy=False):
         return ret(False, msg, None, return_metadata)
 
     # we now know obj is a polars DataFrame or LazyFrame
+    index_cols = get_mi_cols(obj)
+
+    if scitype == "Series":
+        cols_msg = (
+            f"{var_name} must have exactly one index column, "
+            f"found {len(index_cols)}, namely: {index_cols}"
+        )
+        right_no_index_cols = len(index_cols) <= 1
+    elif scitype == "Panel":
+        cols_msg = (
+            f"{var_name} must have exactly two index columns, "
+            f"found {len(index_cols)}, namely: {index_cols}"
+        )
+        right_no_index_cols = len(index_cols) == 2
+    elif scitype == "Hierarchical":
+        cols_msg = (
+            f"{var_name} must have three or more index columns, "
+            f"found {len(index_cols)}, namely: {index_cols}"
+        )
+        right_no_index_cols = len(index_cols) >= 3
+    else:
+        return RuntimeError(
+            'scitype arg of check_dask_frame must be one of strings "Series", '
+            f'"Panel", or "Hierarchical", but found {scitype}'
+        )
+
+    if not right_no_index_cols:
+        return ret(False, cols_msg, None, return_metadata)
+
+    # check if index columns are monotonically increasing
+    if not is_monotonically_increasing(obj, scitype=scitype):
```

Comment:
monotonically increasing is not a requirement for `Table` scitype.

Reply:
Yes we can fix this with a upper if. Also this makes me think should we move this checks to the particular scitype directory in the `_check.py` module. 
I would still prefer this implementation.

Reply:
perhaps move it into another `_check_monotonic` function?